BatchApex/BatchApex.cls

```diff
@@ -0,0 +1,119 @@
+/**
+ * Collection of Closed Won Opportunities 
+ * The Primary contact associated with an account needs to know which all are the 
+ * closed won opportunities for his account per day. 
+ * Create a process which will run everyday in the midnight & collect all the 
+ * Closed won opportunities of that account in a day & send an email to Primary Contact. 
+ * The email body should contain the Opportunity Name & it's respective amount 
+ * with the aggregated amount in the end. 
+ */
+global with sharing class BatchApex implements Database.Batchable<Sobject>, Database.Stateful, 
```

Comment:
we use statefull to maintain the state of the variables, as after the execute block is complete the variables are reset. 
Schedulable is used to schedule a batch.  funcationality will work but we wont get the correct results